mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.purgeOrder = async (req, res) => {
+  try {
+    const result = await Order.findOneAndRemove({ _id: req.params.orderId });
```

Comment:
The `req.params.orderId` should be validated and sanitized before being used in the `Order.findOneAndRemove` method to prevent potential MongoDB injection attacks. Additionally, the `orderId` should be cast to the correct data type (e.g., ObjectId) to ensure correct query execution.